opsdroid/connector/gitter/__init__.py

```diff
@@ -0,0 +1,80 @@
+
+import logging
+import aiohttp
+import asyncio
+import json
+import urllib
+
+from opsdroid.connector import Connector,register_event
+from opsdroid.events import Message
+
+_LOGGER = logging.getLogger(__name__)
+GITTER_STREAM_API = "https://stream.gitter.im/v1/rooms"
+
+
+class ConnectorGitter(Connector):
+
+  def __init__(self, config, opsdroid=None):
+    """Create the connector."""
+    super().__init__(config, opsdroid=opsdroid)
+    _LOGGER.debug("Starting Gitter connector")
+    self.name = "gitter"
+    self.session = None
+    self.response = None
+    self.bot_name = self.config.get("bot-name", 'opsdroid')
+    self.room_id = self.config.get("room-id")
+    self.access_token = self.config.get("access-token")
+    self.update_interval = 1
+    self.opsdroid = opsdroid
+
+  async def connect(self):
+
+    # Create connection object with chat library
+    _LOGGER.debug("Connecting with gitter stream")
+    self.session = aiohttp.ClientSession()
+    gitter_url = self.build_url(GITTER_STREAM_API,self.room_id,"chatMessages", access_token = self.access_token)
+    _LOGGER.debug("Gitter stream url %s",gitter_url )
+    self.response = await self.session.get(gitter_url)
+
+  def build_url(self,base_url , *res, **params):
+    url = base_url
+    for r in res:
+        url = '{}/{}'.format(url, r)
+    if params:
+        url = '{}?{}'.format(url, urllib.parse.urlencode(params))
+    return url
+
+  async def listen(self):
+    _LOGGER.debug("Listening with gitter stream")
+    while True:
+      await asyncio.sleep(self.update_interval)
+      async for data in self.response.content.iter_chunked(1024):
+        _LOGGER.info(data)
+        try:
+            data = await self.parse_message(data)
+            if data !=None:
+              message = Message(
```

Comment:
You're missing the `target` argument here. This should be passed in after the user id. However you're passing in the connector self reference instead. There should be 4 arguments, but you have three. 

This is why `self.connector.configuration` is failing because `self.connector` is defaulting to `None